internal/controllers/addon/phase_ensure_secretpropagation.go

```diff
@@ -18,9 +17,7 @@ import (
 
 // Sub-Reconciler taking care of secret propagation.
 type addonSecretPropagationReconciler struct {
-	cachedClient, uncachedClient client.Client
-	scheme                       *runtime.Scheme
-	addonOperatorNamespace       string
+	addonReconciler *AddonReconciler
 }
```

Comment:
The whole idea behind these sub-reconcilers is to make testing and refactoring easier.
A huge part of this is to separate the dependencies that a particular sub-reconciler needs (Not everything needs the uncached client, not everything accesses the metrics recorder, or needs the addonOperatorNamespace injected)

Now by just giving these sub-reconciler the whole *AddonReconciler to access everything, we don't gain anything anymore IMHO, because it defeats the whole reason for separating the functionality into it's own struct.